### ISSUE TITLE
Introduce new methods to construct subaccount id from an existing AccountIdRef and a prefix

### ISSUE THREAD

**Issue by frol** [near/near-account-id-rs#23]
I see the following two methods to be useful:

* `AccountIdRef::new_subaccount_id(&self, subaccount_name: &str) -> Result<AccountId, ParseAccountError>`
* `const fn AccountIdRef::new_subaccount_id_or_panic(&self, subaccount_name: &str) -> AccountId`

Here is a usage example:

```rust
const ALICE: &AccountIdRef = AccountIdRef::new_or_panic("alice.near");
const ALICE_TEST: &AccountIdRef = ALICE.new_subaccount_id_or_panic("test"); // test.alice.near
```

Good first issues note:

* Make sure to cover the new method with tests and documentation string with usage examples

**Comment by Jujumba** [comment]
I'm a bit confused: `const fn AccountIdRef::new_subaccount_id_or_panic(&self, subaccount_name: &str) -> AccountId` returns `AccountId`, but usage example explicitly states that the return type is `AccountIdRef`. Is it me misseeing something, or there is actually a typo?

@frol

**Comment by frol** [comment]
@Jujumba Where do you see `new_subaccount_id_or_panic` returning `AccountId`?

I expect const-generics to be used at compile time to concatenate the str slices. Here is the crate I would explore using: https://crates.io/crates/constcat

**Comment by Jujumba** [comment]
> @Jujumba Good catch! I have just fixed the signature. I expect const-generics to be used at compile time to concatenate the str slices. Here is the crate I would explore using: https://crates.io/crates/constcat

I'm not sure that it's possible to concatenate these slices at const context. `constcat` crate requires variables to be defined as `const` and const-generics don't allow `&str` type.

@frol

**Comment by frol** [comment]
>  const-generics don't allow `&str` type.

@Jujumba That is not true: https://docs.rs/near-account-id/latest/src/near_account_id/account_id_ref.rs.html#84-88

**Comment by Jujumba** [comment]
> > const-generics don't allow `&str` type.
> 
> @Jujumba That is not true: https://docs.rs/near-account-id/latest/src/near_account_id/account_id_ref.rs.html#84-88

Oh, you meant this. Then you are absolutely right. I thought of the following:
```rust
pub fn function<const S: &'static str>() {
}
```

**Comment by frol** [comment]
#23 introduced the new method `sub_account()` on any AccountIdRef. While there is no `const` method there, I will close this issue as done and will open a new one for `const` method once I will see a need for it (so far, it looks like it is not demanded)
